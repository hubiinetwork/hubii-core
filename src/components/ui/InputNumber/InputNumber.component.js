import React from 'react';
import PropTypes from 'prop-types';
import Input from '../Input';

export default class InputNumber extends React.PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      value: props.minValue,
    };
    this.onChange = this.onChange.bind(this);
  }
  onChange(event) {
    const { minValue, maxValue } = this.props;

    if (event.target.value < minValue || event.target.value > maxValue) {
      return;
    }
    this.setState({
      value: event.target.value,
    });
  }
  render() {
    const { ...rest } = this.props;
    return (
      <Input
        value={this.state.value}
        onChange={this.onChange}
        type={'number'}
        {...rest}
      />
    );
  }
  }

InputNumber.propTypes = {
  minValue: PropTypes.number,
  maxValue: PropTypes.number,
};

