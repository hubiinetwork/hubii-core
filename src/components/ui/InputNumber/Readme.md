Sample of Inputnumber:

```js
<div style={{ padding: 25, backgroundColor: '#26404D' }}>
  <InputNumber
  minValue = {0}
  maxValue = {1000}
  />
</div>
```
