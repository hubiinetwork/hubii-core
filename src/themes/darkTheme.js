// ****** white ****** //

const white = '#ffffff';

// ****** teal ****** //

const teal = '#3b6174';
const tealDark = '#263c46';
const tealLight = '#26404d';
const tealLighter = '#395664';
const tealDarker = '#2f4d5c';
const teal50 = 'rgba(47,77,92,.5)';

// ****** seaGreen ****** //

const seaGreen = '#3df5cd';
const seaGreenLight = 'rgba(80,227,194,.05)';
const seaGreen50 = 'rgba(80,227,194,.5)';

// ****** grey ****** //

const grey = 'rgba(255,255,255,0.5)';
const greyDim = 'rgba(255,255,255,0.3)';
const greyLight = '#C0CDD3';
const greyDark = '#43616F';
const greyDarker = '#445f6c';
const grey50 = 'rgba(140,165,177,.5)';
const grey8C = '#8CA5B1';

// ****** black ****** //

const black = '#122b2b';
const black50 = 'rgba(36, 59, 71, 0.5)';
const blackLight = '#27404c';
const blackLighter = '#404f56';

// ****** green ****** //

const green = 'greenyellow';
const greenLight = 'rgba(173, 255, 47, 0.5)';

// ****** yellow ****** //

const orange = 'orange';

// ****** shadows ****** //

const shadowLight = '0 2px 5px 0 rgba(0, 0, 0, 0.15)';
const shadowDark = '0 3px 7.5px 0 rgba(0, 0, 0, 0.15)';

export default {
  iconColor: '',
  mode: 'dark',
  color: 'red',
  shadows: {
    light: shadowLight,
    dark: shadowDark
  },
  palette: {
    primary: teal,
    primary1: tealLight,
    primary2: tealLighter,
    primary3: tealDark,
    primary4: tealDarker,
    primary5: teal50,

    secondary: grey,
    secondary1: greyLight,
    secondary2: greyDark,
    secondary3: greyDarker,
    secondary4: grey50,
    secondary5: greyDim,
    secondary6: grey8C,

    dark: black,
    dark1: blackLight,
    dark2: blackLighter,
    dark3: black50,

    light: white,

    info: seaGreen,
    info1: seaGreenLight,
    info2: seaGreen50,

    success: green,
    success1: greenLight,
    warning: orange
  }
};
