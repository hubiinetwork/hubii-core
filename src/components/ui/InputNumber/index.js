import InputNumber from './InputNumber.component';
export default InputNumber;
