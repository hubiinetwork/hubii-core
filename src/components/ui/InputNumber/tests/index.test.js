import React from 'react';
import { shallow, mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import InputNumber from '../index';
import darkTheme from '../../../../themes/darkTheme';

describe('<InputNumber />', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<InputNumber handleChange={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should have correct props', () => {
    const wrapper = shallow(<InputNumber minValue={20} maxValue={30} handleChange={() => {}} />);
    const wrapperProps = wrapper.instance().props;
    expect(wrapperProps.minValue).toEqual(20);
    expect(wrapperProps.maxValue).toEqual(30);
  });
  it('onchange should run', () => {
    const component = mount(
      <ThemeProvider theme={darkTheme}>
        <InputNumber handleChange={() => {}} />
      </ThemeProvider>
    );
    const input = component.find('.ant-input');
    input.simulate('change');
    expect(input.props().className).toHaveLength(9);
  });
  it('should render correctly', () => {
    const component = mount(
      <ThemeProvider theme={darkTheme}>
        <InputNumber handleChange={() => {}} />
      </ThemeProvider>
    );
    const input = component.find('.ant-input');
    expect(input.text()).toBe('');
  });
});
