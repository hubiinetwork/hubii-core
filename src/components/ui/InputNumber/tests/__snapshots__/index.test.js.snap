// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputNumber /> should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InputNumber
    handleChange={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "defaultValue": undefined,
      "max": undefined,
      "min": undefined,
      "onChange": [Function],
      "type": "number",
      "value": undefined,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "defaultValue": undefined,
        "max": undefined,
        "min": undefined,
        "onChange": [Function],
        "type": "number",
        "value": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
